core/common/src/main/java/alluxio/underfs/options/CreateOptions.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.options;
+
+import alluxio.Configuration;
+import alluxio.annotation.PublicApi;
+import alluxio.security.authorization.Permission;
+
+import com.google.common.base.Objects;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Method options for creating a file in UnderFileSystem.
+ */
+@PublicApi
+@NotThreadSafe
+public final class CreateOptions {
+  // Permission to set for the file being created.
+  private Permission mPermission;
+
+  /**
+   * Constructs a default {@link CreateOptions}.
+   */
+  public CreateOptions() {
+    mPermission = Permission.defaults();
+  }
```

Comment:
add a TODO to switch this to a single `defaults()` once / if we move to singleton configuration